### PR TITLE
chore(python): more precise checks for inefficient apply warnings

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Checking that the parameter name is in the right place because I'm paranoid

### CODE REVIEW

**Comment by alexander-beedie** [comment]
> Checking that the parameter name is in the right place because I'm paranoid

_"Just because you're paranoid it doesn't mean they're **not** after you..." (Catch-22)_

**Comment by alexander-beedie** [comment]
If I'm understanding this correctly, you want to double-check that there is not a known param name associated with `LOAD_CONST`? (Or perhaps better-stated the other way round). 

I don't really like all the extra copy/paste boilerplate going on here for something that I'm not sure is a real problem - do you have any concrete examples where this would find something? 🤔

**Comment by MarcoGorelli** [comment]
It's the other way round - if it's `LOAD_FAST`, I'd like to check that it's the expected variable which is getting loaded

But you're right, I don't have any concrete examples, I'm just (perhaps unnecessarily) worried that we're missing something. Happy to keep this out and only add it if bugs are reported if you prefer though

**Comment by MarcoGorelli** [comment]
I've pushed a different solution to increasing defensiveness / addressing my paranoia: checking more precisely within `_matches`, by taking `opnames: list[set[str]]` rather than `opnames: list[str]`. Like this, we can get rid of `LOAD_*` and only check `LOAD_CONST` and `LOAD_FAST`, and be sure that no `LOAD_DEREF` is gonna sneak in there and catch us off-guard

**Comment by alexander-beedie** [comment]
> I've pushed a different solution to increasing defensiveness / addressing my paranoia

I _much_ prefer this one  ;)

Was wondering if there would be a performance impact from using a set membership check  instead of string comparison, but actually it turns out there is _zero_ impact! (Which is great).

File: py-polars/polars/utils/udfs.py [hunk 1]
```diff
@@ -536,12 +536,7 @@ def _matches(
         n_required_ops, argvals = len(opnames), argvals or []
         instructions = self._instructions[idx : idx + n_required_ops]
         if len(instructions) == n_required_ops and all(
-            any(
-                inst.opname == match_opname
-                if match_opname[-1] != "*"
-                else inst.opname.startswith(match_opname[:-1])
-                for match_opname in match_opnames
-            )
+            any(inst.opname == match_opname for match_opname in match_opnames)
```
Comment (alexander-beedie) [inline py-polars/polars/utils/udfs.py:539]: Isn't this just `inst.opname in match_opnames` now?

Comment (MarcoGorelli) [reply py-polars/polars/utils/udfs.py]: thanks!

**Review by alexander-beedie — APPROVED** [review]

**Pull request merged**